### PR TITLE
Consolidate Button styles in new `ButtonStyles` class

### PR DESCRIPTION
**Pull request by MichaelZetune**
### Description

Previously button styles were located in `ApplicantStyles.java`. They were not well organized, and had many style variables to represent the same style in many places. Here, we consolidate everything into `ButtonStyles.java`.

There are two main button styles: `ButtonStyles.SOLID_WHITE` and `ButtonStyles.SOLID_BLUE` and some minor ones. The most important part of this refactor is that button styles are no longer dedicated to particular use cases; they should be generalizable.

#### General

- [x] Add `ButtonStyles.java`
- [x] Move button styles from `ApplicantStyles.java` to that new class
- [x] Consolidate redundant styles
   - Remove `BUTTON_ENUMERATOR_ADD_ENTITY`, `BUTTON_ENUMERATOR_REMOVE_ENTITY`, `BUTTON_TI_DASHBOARD`, `BUTTON_SUBMIT_APPLICATION`, `BUTTON_PROGRAM_APPLY` - these are per-use-case styles and shouldn't be represented in `ButtonStyles`. Instead, we use the common styles in the places where they're needed and add custom style classes in the particular instances where required.

#### User visible changes

None; only code changes.

#### New Features

None.

### Issue(s) this completes

Relates to #4657.

### CODE REVIEW

File: server/app/views/components/ButtonStyles.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * A collection of styles for buttons throughout CiviForm. Importantly, the public style Strings in
+ * this file should be **generalizable**, meaning they are not designed for a particuar use case.
+ * Styles should not be called i.e. BUTTON_SUBMIT_APPLICATION for example. If a specific change is
+ * required for a use case, use a general style and add more classes in the client code.
+ */
+public final class ButtonStyles {
+  /**
+   * Base styles for buttons in the applicant UI. This is missing a specified text size, so that
```
Comment (reemaka) [inline server/app/views/components/ButtonStyles.java:14]: is this still specific to the applicant UI or is this comment outdated now?

File: server/app/views/components/ButtonStyles.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   * should be added by other button style constants that use this as a base.
+   */
+  private static final String BUTTON_BASE =
+      StyleUtils.joinStyles(
+          "block", "py-2", "text-center", "rounded-full", "border", "border-transparent");
+
+  /** Base styles for buttons with a solid background color. */
```
Comment (reemaka) [inline server/app/views/components/ButtonStyles.java:21]: **optional:** maybe specify it's solid blue specifically

File: server/app/views/components/ButtonStyles.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          BUTTON_BASE,
+          BaseStyles.BG_SEATTLE_BLUE,
+          "text-white",
+          "rounded-full",
+          StyleUtils.hover("bg-blue-700"),
+          StyleUtils.disabled("bg-gray-200", "text-gray-400"));
+
+  /** Base styles for semibold buttons with a solid background. */
+  private static final String BUTTON_BASE_SOLID_SEMIBOLD =
+      StyleUtils.joinStyles(BUTTON_BASE_SOLID, "font-semibold", "px-8");
```
Comment (reemaka) [inline server/app/views/components/ButtonStyles.java:33]: **optional:** BUTTON_BASE_SOLID is only used in BUTTON_BASE_SOLID_SEMIBOLD, so you could consider combining those unless you think there will be a use case for a non-semibold version of it in the future.

Same for BUTTON_BASE_OUTLINE and BUTTON_BASE_OUTLINE_SEMIBOLD

File: server/app/views/applicant/ApplicantInformationView.java [hunk 4]
```diff
@@ -80,7 +81,7 @@ public Content render(
 
     String submitText = messages.at(MessageKey.BUTTON_UNTRANSLATED_SUBMIT.getKeyName());
     ButtonTag formSubmit =
-        submitButton(submitText).withClasses(ApplicantStyles.BUTTON_SELECT_LANGUAGE);
+        submitButton(submitText).withClasses(ButtonStyles.SOLID_BLUE, "text-base", "mx-auto");
```
Comment (reemaka) [inline server/app/views/applicant/ApplicantInformationView.java:84]: Looks like SOLID_BLUE already has the "text-base" class?

Comment (MichaelZetune) [reply server/app/views/applicant/ApplicantInformationView.java:84]: Done.

File: server/app/views/components/ButtonStyles.java (hunk 2 above)
Comment (MichaelZetune) [reply server/app/views/components/ButtonStyles.java:21]: Done.

File: server/app/views/components/ButtonStyles.java (hunk 3 above)
Comment (MichaelZetune) [reply server/app/views/components/ButtonStyles.java:33]: Good catch - consolidated.

File: server/app/views/questiontypes/EnumeratorQuestionRenderer.java [hunk 5]
```diff
@@ -87,7 +87,11 @@ protected DivTag renderInputTags(
                     .withType("button")
                     .condAttr(hasErrors, "aria-invalid", "true")
                     .withClasses(
-                        ApplicantStyles.BUTTON_ENUMERATOR_ADD_ENTITY,
+                        ButtonStyles.SOLID_BLUE,
+                        "text-base",
```
[the remark this answers is not in the document]
Comment (MichaelZetune) [reply server/app/views/questiontypes/EnumeratorQuestionRenderer.java:91]: Done.

**Review by reemaka — APPROVED** [review]

**Pull request merged**